### PR TITLE
Add java examples

### PR DESCRIPTION
**Pull request by PaperHS**

### CODE REVIEW

File: examples/javademo/src/main/java/ConsumeAsset.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Author       : shan@arcblock.io
+ * Time         : 2019-11-29
+ * Edited By    :
+ * Edited Time  :
+ * Description  :
+ **/
+class ConsumeAsset extends BaseConfig {
+
+  public static void main(String[] args){
+    ForgeSDK forge = ForgeSDK.Companion.connect("localhost", BaseConfig.serverPort);
```
Comment (wangshijun) [inline examples/javademo/src/main/java/ConsumeAsset.java]: Why do we nest `.connect` under `ForgeSDK`, can the developer just call `ForgeSDk.connect`

File: examples/javademo/src/main/java/ComsumeAssetGateKeeper.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+class ComsumeAssetGateKeeper extends BaseConfig {
+  public static void main(String[] args){
+    ForgeSDK forge = ForgeSDK.Companion.connect("localhost", BaseConfig.serverPort);
+    Rpc.ResponseSendTx response;
+    Type.ChainInfo chainInfo = forge.getChainInfo().getInfo();
+
+    WalletInfo Issuer = forge.createWallet();
+    response = forge.declare("Issuer", Issuer);
+    WalletInfo GateKeeper = forge.createWallet();
+    response =forge.declare("GateKeeper", GateKeeper, Issuer.getAddress());
```
Comment (wangshijun) [inline examples/javademo/src/main/java/ComsumeAssetGateKeeper.java]: looks like we are not enforcing coding format through out the code

File: examples/javademo/src/main/java/ComsumeAssetGateKeeper.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * Edited Time  :
+ * Description  :
+ **/
+class ComsumeAssetGateKeeper extends BaseConfig {
+  public static void main(String[] args){
+    ForgeSDK forge = ForgeSDK.Companion.connect("localhost", BaseConfig.serverPort);
+    Rpc.ResponseSendTx response;
+    Type.ChainInfo chainInfo = forge.getChainInfo().getInfo();
+
+    WalletInfo Issuer = forge.createWallet();
```
Comment (wangshijun) [inline examples/javademo/src/main/java/ComsumeAssetGateKeeper.java]: about variable naming, looks like we do not have a fixed pattern

File: examples/javademo/src/main/java/ComsumeAssetGateKeeper.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    response = forge.declare("Issuer", Issuer);
+    WalletInfo GateKeeper = forge.createWallet();
+    response =forge.declare("GateKeeper", GateKeeper, Issuer.getAddress());
+    WalletInfo Consumer = forge.createWallet();
+    response =forge.declare("Consumer", Consumer);
+
+    forge.poke(Consumer);// consumer get money
+
+    //create Asset for Thomas
+    Result result = forge.createAsset("json",("{\"a\":"+ UUID
```
Comment (wangshijun) [inline examples/javademo/src/main/java/ComsumeAssetGateKeeper.java]: json.encode makes it more readable?

File: examples/javademo/src/main/java/ComsumeAssetGateKeeper.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      .randomUUID().toString() +"}").getBytes(), "testAsset", Issuer);
+    response = result.getResponse();//create asset transaction response
+    String assetAddress = result.getAddress();
+
+
+    //wait for block commit
+    try {
```
Comment (wangshijun) [inline examples/javademo/src/main/java/ComsumeAssetGateKeeper.java]: repetitive code, why not in a helper function

File: examples/javademo/src/main/java/ConsumeAsset.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import java.util.UUID;
+
+import forge_abi.Rpc;
+import io.arcblock.forge.ForgeSDK;
+import io.arcblock.forge.Result;
+import io.arcblock.forge.did.WalletInfo;
+import io.arcblock.forge.extension.BigIntegerExt;
+
+/**
+ * █████╗ ██████╗  ██████╗██████╗ ██╗      ██████╗  ██████╗██╗  ██╗
```
Comment (wangshijun) [inline examples/javademo/src/main/java/ConsumeAsset.java]: I think this logo can be removed

File: examples/javademo/src/main/java/DelegateWithMultiSig.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    // alice get money
+    forge.poke(alice);
+    //create Asset for Thomas
+    Result result = forge.createAsset("json",("{\"a\":"+ UUID
+      .randomUUID().toString() +"}").getBytes(), "testAsset", Thomas);
+    response = result.getResponse();//create asset transaction response
+    String assetAddress = result.getAddress();
+
+    //prepare Transaction
```
Comment (wangshijun) [inline examples/javademo/src/main/java/DelegateWithMultiSig.java]: Let's make `prepare` and `finalize` api consistent with elixir sdk?

- `prepareExchange`
- `finalizeExchange`

File: examples/javademo/src/main/java/ConsumeAsset.java (hunk 1 above)
Comment (PaperHS) [reply examples/javademo/src/main/java/ConsumeAsset.java]: Java call Kotlin  Object method grammar

File: examples/javademo/src/main/java/DelegateWithMultiSig.java (hunk 7 above)
Comment (PaperHS) [reply examples/javademo/src/main/java/DelegateWithMultiSig.java]: They has some distinctions, I will consider it next version

**Pull request merged**